packages/nimble-components/src/rich-text-mention/users/models/rich-text-mention-users-validator.ts

```diff
@@ -0,0 +1,39 @@
+import type { Mapping } from '../../../mapping/base';
+import type { MentionHref } from '../../../mapping/base/types';
+import { MappingMentionUser } from '../../../mapping/mention-user';
+import type { MentionInternals } from '../../base/models/mention-internals';
+import {
+    baseValidityFlagNames,
+    RichTextMentionValidator
+} from '../../base/models/mention-validator';
+
+const usersValidityFlagNames = [
+    ...baseValidityFlagNames,
+    'missingDisplayNameValue'
+] as const;
+
+/**
+ * Validator for RichtextMentionUsers
+ */
+export class RichTextMentionUsersValidator extends RichTextMentionValidator<
+    typeof usersValidityFlagNames
+> {
+    public constructor(columnInternals: MentionInternals<unknown>) {
+        super(columnInternals, usersValidityFlagNames, [MappingMentionUser]);
+    }
+
+    public override validate(
+        mappings: Mapping<MentionHref>[],
+        pattern: string
+    ): void {
+        super.validate(mappings, pattern);
+        this.validateNoMissingDisplayName(mappings);
+    }
+
+    private validateNoMissingDisplayName(mappings: MappingMentionUser[]): void {
+        const invalid = mappings.some(
```

Comment:
I think this will fail if there are no mappings but we want to support that case, for example while the app is fetching users to map.

When you write tests please add coverage for that case.